File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
import math
class Solution:
    def longestPalindrome(self, s: str) -> str:
        l = len(s)
        while l > 1:
            for i in range(len(s) - l + 1):
                target = s[i:i+l]

                for j in range(math.ceil(len(target)/2)):
                    if target[j] != target[(-1 * j) - 1]:
                        break
                    
                    if j == math.ceil(len(target)/2) - 1:
                        return target
            l -= 1
        return s[0]

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        if (s.length() < 2 || new StringBuffer(s).reverse().toString().equals(s) ){
            return s;
        }

        int start = 0, end = 0;
        for (int i = 0; i < s.length()-1; i++){
            int[] result = checkPalindromic(i, i, s);
            if (end - start < result[1] - result[0]){
                start = result[0];
                end = result[1];
            }
            if (s.charAt(i) == s.charAt(i+1)){
                result = checkPalindromic(i, i+1, s);
                if (end - start < result[1] - result[0]){
                    start = result[0];
                    end = result[1];
                }
            }    
        }
        return s.substring(start, end + 1);
    }

    private int[] checkPalindromic(int l, int r, String s){
        int left = l, right = r;
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)){
            left -= 1;
            right += 1;
        }
        return new int[]{left+1, right-1};
    }   
}